Compfiles/Imo2012P4.lean

```diff
@@ -20,11 +20,308 @@ the following equality holds:
 
 namespace Imo2012P4
 
-determine solution_set : Set (ℤ → ℤ) := sorry
+def odd_const : Set (ℤ → ℤ) := fun f =>
```

Comment:
`odd_const`, `mod4_cycle`, `square_set`, `sub_sq'`, `Int.lt_of_ns_lt_ns` and `myInduction` should be inside of `snip begin ... snip end`. https://github.com/dwrensha/compfiles/blob/7b57a0dca9e0248ce1bfbf0ea8a4aebc7db9e72c/ProblemExtraction.lean#L187-L191

For example: https://github.com/dwrensha/compfiles/blob/7b57a0dca9e0248ce1bfbf0ea8a4aebc7db9e72c/Compfiles/Imo1965P2.lean#L33-L72

Reply:
This will make it so that the problem web page (https://dwrensha.github.io/compfiles/problems/Compfiles.Imo2012P4.html) does not show any information about the solution.

Reply:
done!